src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -156,47 +156,72 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
             }
             catch (JobConflictException)
             {
-                // The job was updated by another process.
-                _logger.LogWarning("The job was updated by another process.");
-
-                // TODO: We will want to get the latest and merge the results without updating the status.
-                return;
+                await HandleJobConflictAsync(cancellationToken);
             }
             catch (Exception ex)
             {
                 // The job has encountered an error it cannot recover from.
                 // Try to update the job to failed state.
                 _logger.LogError(ex, "Encountered an unhandled exception. The job will be marked as failed.");
 
-                await UpdateAndCommitJobStatus(OperationStatus.Failed, updateEndTimestamp: true, cancellationToken);
+                await CompleteJobAsync(OperationStatus.Failed, cancellationToken);
             }
         }
 
-        private async Task UpdateAndCommitJobStatus(OperationStatus operationStatus, bool updateEndTimestamp, CancellationToken cancellationToken)
+        private async Task CompleteJobAsync(OperationStatus completionStatus, CancellationToken cancellationToken)
         {
-            _exportJobRecord.Status = operationStatus;
-
-            if (updateEndTimestamp)
-            {
-                _exportJobRecord.EndTime = Clock.UtcNow;
-            }
+            _exportJobRecord.Status = completionStatus;
+            _exportJobRecord.EndTime = Clock.UtcNow;
 
-            await UpdateAndCommitJobRecord(_exportJobRecord, cancellationToken);
+            await UpdateJobRecordAsync(cancellationToken);
         }
 
-        private async Task UpdateAndCommitJobRecord(ExportJobRecord jobRecord, CancellationToken cancellationToken)
+        private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
         {
             using (IScoped<IFhirOperationDataStore> fhirOperationDataStore = _fhirOperationDataStoreFactory())
             {
-                ExportJobOutcome updatedExportJobOutcome = await fhirOperationDataStore.Value.UpdateExportJobAsync(jobRecord, _weakETag, cancellationToken);
+                ExportJobOutcome updatedExportJobOutcome = await fhirOperationDataStore.Value.UpdateExportJobAsync(_exportJobRecord, _weakETag, cancellationToken);
 
```

Comment:
This could still throw a `JobConflictException` since we will be updating the JobRecord during this stage.